app/models/manageiq/providers/azure_stack/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,171 @@
+class ManageIQ::Providers::AzureStack::Inventory::Collector::TargetCollection < ManageIQ::Providers::AzureStack::Inventory::Collector
+  require_nested :V2018_03_01
+  require_nested :V2017_03_09
+
+  def initialize(manager, target)
+    super
+
+    parse_targets!
+    infer_related_ems_refs!
+
+    # Reset the target cache, so we can access new targets inside
+    target.manager_refs_by_association_reset
+  end
+
+  # ##################################
+  # Target collection for CloudManager
+  # ##################################
+  def resource_groups
+    return @resource_groups unless @resource_groups.nil?
+    return [] if (refs = references(:resource_groups)).blank?
```

Comment:
Really minor because this isn't doing an API call, but if `references(:resource_groups)` is empty and `@resource_groups` isn't set to `[]` then subsequent calls to `resource_groups` will skip the guard and hit this line repeatedly.

This could be simplified by doing `refs = references(:resource_groups) || []` so that `[].map { ... }` would set `@resource_groups = []` below

Reply:
Done, thanks :+1: